docs/developer/design/authentication.rst

```diff
@@ -0,0 +1,154 @@
+Authentication
+==============
+
+Todo
+----
+
+- Provide uml diagrams for Dex + MetalK8s authentication flow
+
+Open questions
+--------------
+
+- Is support for Dex monitoring and alerting on Grafana
+  something we should think about now?
+
+Abstract
+--------
+
+To be able to identify who a user is and to manage(Create/Edit/Delete)
+these user identities, MetalK8s needs a middleman that acts in this light
+as an Identity Provider.
+
+To accomplish this, we introduce the concept of an OpenID Connect identity
+provider which has as role identity federation in a typical MetalK8s cluster.
```

Comment:
Not exactly why we do this. To me, the rationale can be summed up in three points:
- we already have authentication with K8s API, but it's static declaration is cumbersome, which explains why we want a dynamic IdP to replace it
- we also want to integrate with existing IdPs a customer would like to plug in
- finally, and most importantly, we don't want to store user credentials, which is where OIDC (or OAuth, for that matter), come into play